### ISSUE TITLE
Model stops converging after safe log commit?

### ISSUE THREAD

**Issue by Forevian** [magenta/ddsp#191]
I had a model where I had good results resynthesizing various acoustic percussive hits with noises and non-harmonic sinusoids. The model stopped converging after this commit:
https://github.com/magenta/ddsp/commit/db2c3c1e581aa2adddef3af25656621cbada917e
Any tips what could possibly cause any issues here?

**Comment by jesseengel** [comment]
I'm surprised to hear that. the only real difference is replacing `log(x + 1e-5)` with `log(where( x <= 0, 1e-5, x)`. I guess that would mean if you had a lot of very very quiet audio (such as sparse percussive hits) the earlier version would put a threshold on the loss and wash out the differences. 

Can you confirm if you replace the safe log with the earlier version (locally) does it converge again?

**Comment by Forevian** [comment]
Yes, I can confirm that if I use 0.12.0 and change the above back in spectral_ops.py, then I get the expected result. I am trying to overfit on a single sample. Not sure if I understand why this tiny offset value would change the behavior of the network, can you rephrase this? Btw, the sample looks like this:
![image](https://user-images.githubusercontent.com/4150342/90748612-b40d4700-e2d2-11ea-9737-9f03456fc352.png)

**Comment by jesseengel** [comment]
That's really interesting. Cool sound btw too :). Can you check if this works for you? 

```python
def safe_log(x, eps=1e-5):
  return log(where( x <= eps, eps, x)
```

I'll try again. In other words, this safe log is used in the log_magnitude loss functions. previously, in areas of mag < 1e-5, the differences would be washed out by adding 1e-5, so the loss would be 0 because 1e-5 was added to both sides. But with the where statement as it is in 0.12 that only happens for 0 or < 0, which means that the loss function is caring a lot about quite bits that are inaudible (because it's on a log scale).

**Comment by jesseengel** [comment]
Sorry it took sooooo long to make a 3 character PR :).

**Comment by Forevian** [comment]
Hey, I should have done that myself, but I had to focus on non-ddsp related tasks ever since we've discussed this. Thanks for your great work, we all appreciate it!
